core/src/main/java/org/geysermc/geyser/entity/type/ThrowableItemEntity.java

```diff
@@ -48,6 +48,10 @@ public ThrowableItemEntity(GeyserSession session, int entityId, long geyserId, U
         super(session, entityId, geyserId, uuid, definition, position, motion, yaw, pitch, headYaw);
         setFlag(EntityFlag.INVISIBLE, true);
         invisible = false;
+        if (session.isFrozen()) {
+            age = 4;
+            checkVisibility();
+        }
     }
```

Comment:
Is the age set to `4` here so that `checkVisibility()` sets the entity back to visible? or i don't understand

Reply:
When ticks are disabled, the entity is not updated resulting in it staying invisble

Reply:
The behaviour is still different from java, as the throwable checks how close the player is to the entity, not based on age, so even if the state is frozen it will still be hidden if its close to you. I think this should be changed to update its invisibility status anyway

Reply:
> the throwable checks how close the player is to the entity, not based on age

`(age >= 4 || position.distanceSquared(playerPos) > 16)` checks both? or we are talking about something different.

Reply:
I rewrote most of the code to mimic java with throwables when the game is frozen in latest commit